### PR TITLE
Add SecureContext and API behavior of browsers

### LINKED ISSUE #25: Restrict interfaces to secure contexts?
In [Intent to Implement and Ship: Remove [NoInterfaceObject] from Geolocation API interfaces](https://groups.google.com/a/chromium.org/d/msg/blink-dev/Xig9oewsQMA/TrVKGUKGAAAJ) @mikewest asks:
> We limit usage of the Geolocation API to secure contexts. While we don't mark `navigator.geolocation` as `[SecureContext]`, it seems reasonable to mark these new interfaces as such. Is that a change we could bring into the spec?

This seems reasonable to me. However, a hindrance is that the spec currently doesn't mention secure contexts at all. http://w3c.github.io/geolocation-api/#dom-navigatorgeolocation-getcurrentposition doesn't even seem to have a blanket "reject the request if you feel like it" allowance that some specs do, seemingly requiring the API to work on non-secure contexts.

To *just* add `[SecureContext]` would leave things in a pretty strange state, so perhaps the algorithms should be modified at the same time to reject requests for non-secure contexts.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #25

The following tasks have been completed:

 * [x] [Modified Web platform tests](https://github.com/web-platform-tests/wpt/pull/18715)

Implementation commitment:

 * [x] WebKit (https://bugs.webkit.org/show_bug.cgi?id=201221)
 * [X] Chromium - all tests pass
 * [X] Gecko - all tests pass

***
<a href="https://pr-preview.s3.amazonaws.com/marcoscaceres/geolocation-api/pull/34.html" title="Last updated on Aug 28, 2019, 11:58 PM UTC (78a4fff)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/34/eea2561...marcoscaceres:78a4fff.html" title="Last updated on Aug 28, 2019, 11:58 PM UTC (78a4fff)">Diff</a>

### CODE REVIEW

**Comment by anssiko** [comment]
I think this PR would benefit from @foolip's review.

**Comment by marcoscaceres** [comment]
Probably want a nod from @annevk too :)

**Comment by annevk** [comment]
This doesn't end up invoking the failure callback.

**Comment by marcoscaceres** [comment]
Will add. Totally forgot to check that 🙃

**Comment by reillyeon** [comment]
For reference, in Blink the failure callback is called with a PositionError where code is PERMISSION_DENIED.

**Comment by marcoscaceres** [comment]
I’ll add some tests to see what we get across browsers.

**Comment by marcoscaceres** [comment]
Updated spec text and added tests https://github.com/web-platform-tests/wpt/pull/18715

 * Chrome and Firefox are aligned
 * Safari returns error code 2 instead of 1.

**Comment by foolip** [comment]
idlharness.js tests will be updated by https://github.com/web-platform-tests/wpt/pull/18717 after the spec change is merged.

**Comment by marcoscaceres** [comment]
Forgot to push before 🤪 should be good to review now.

**Comment by marcoscaceres** [comment]
Eventually, we need to go through and rewrite all these algorithms properly.

File: index.html [hunk 1]
```diff
@@ -428,10 +445,25 @@ <h3><dfn>Geolocation</dfn> interface</h3>
...
+            Optionally, inform the developer that the API can only be used from <a>secure contexts</a>.
+          </li>
+          <li>
+            Terminate this algorithm.
+          </li>
+        </ol>
+      </li>
+    <li>
+   <p>
+   Otherwise, immediately return a long value that uniquely identifies a <a>watch
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply index.html]: Ah! oops. Fixing.

**Comment by marcoscaceres** [comment]
@cdumez, awesome! thanks for fixing it so quickly!

**Comment by marcoscaceres** [comment]
@reillyeon could you review the tests also? Then we should be good to land.

**Comment by marcoscaceres** [comment]
@anssiko could you please review the tests?

**Comment by anssiko** [comment]
@Honry PTAL

**Comment by marcoscaceres** [comment]
Ok, tests merged. Please merge this when possible. @anssiko, could you grant me write acceptance?

**Comment by anssiko** [comment]
Thanks @marcoscaceres for championing this fix and others for updating the implementations, we're ready to land now. @marcoscaceres you have been granted write.

**Pull request merged**